### PR TITLE
W3C plugin

### PR DESCRIPTION
**Pull request by jsirianni**
I can help test this. It depends on a release that does not exists (yet) Stanza v1.1.9. You will need to comment `min_stanza_version` for now. You can test with a dev branch of stanza, I can help set you up.

The W3C plugin wraps the CSV parser, making use of dynamic field names
- File input detects W3C headers and attaches them as labls
- CSV operator will use the `Fields` label to determine the CSV field names dynamically, for every entry.

This allows the user to read arbitrary W3C files without specifying the field names.

### CODE REVIEW

**Comment by jsirianni** [comment]
You can test this plugin by building this branch https://github.com/observIQ/stanza/pull/404

File: plugins/w3c.yaml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    label: Start At
+    description: Start reading file from 'beginning' or 'end'
+    type: enum
+    valid_values:
+      - beginning
+      - end
+    default: end
+  - name: include_file_name
+    label: Include File Name
+    description: Include File Nath as a label
```
Comment (ericwholt) [inline plugins/w3c.yaml:44]:
```suggestion
    description: Include File Name as a label
```

File: plugins/w3c.yaml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+# {{ if .exclude_file_log_path }}
+    exclude:
+  # {{ range $i, $efp := .exclude_file_log_path  }}
+      - '{{ $efp }}'
+  # {{ end }}
+# {{ end }}
+# {{ if $encoding }}
+    encoding: '{{ $encoding }}'
+# {{ end }}
+    include_file_name: true
```
Comment (ericwholt) [inline plugins/w3c.yaml:111]: Duplicate parameter.
```suggestion
```

**Review by ericwholt — CHANGES_REQUESTED** [review]
Using the following config:
``` yaml
    - type: w3c
      file_log_path: 
      - /Users/ericholt/bindplane/bplogagent-dev/msiis-logs/iis.log
      header_delimiter: " "
      delimiter: " "
      start_at: beginning
```
IIS Logs Output:
``` json
{
  "timestamp": "2021-08-23T21:48:43.197971-04:00",
  "severity": 0,
  "labels": {
    "Date": "2020-11-03 16:13:15",
    "Software": "Microsoft Internet Information Services 10.0",
    "Version": "1.0",
    "file_name": "iis.log",
    "log_type": "w3c",
    "plugin_id": "w3c"
  },
  "record": {
    "c-ip": "10.0.0.2",
    "cs(Referer)": "-",
    "cs(User-Agent)": "Go-http-client/1.1",
    "cs-method": "GET",
    "cs-uri-query": "-",
    "cs-uri-stem": "/status",
    "cs-username": "-",
    "date": "2020-11-03",
    "s-ip": "10.0.0.1",
    "s-port": "80",
    "sc-status": "404",
    "sc-substatus": "0",
    "sc-win32-status": "2",
    "time": "16:13:15",
    "time-taken": "1"
  }
}
```

**Review by jhowellbm — APPROVED** [review]

**Pull request merged**